webapp/shop/decorators.py

```diff
@@ -152,7 +152,14 @@ def decorated_function(*args, **kwargs):
                     return flask.redirect(f"/login?next={redirect_path}")
 
             ua_contracts_api = get_ua_contracts_api_instance(
-                user_token, response, session, flask.request.remote_addr
+                user_token,
+                response,
+                session,
+                (
+                    flask.request.headers.getlist("X-Forwarded-For")[0]
+                    if flask.request.headers.getlist("X-Forwarded-For")
+                    else flask.request.remote_addr
+                ),
```

Comment:
Because of the way Flask works, this will send the backend all of the `X-Forwarded-For` values concatenated, so the backend might receive something like `<ip1>, <ip2>,<ip3>`, which is fine.